File: app/src/main/java/com/app/citycareservice/EditProfileActivity.java
package com.app.citycareservice;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.util.Patterns;
import android.view.View;
import android.widget.ImageView;

import androidx.appcompat.app.AppCompatActivity;

import com.app.citycareservice.service.UpdateProfileService;
import com.app.citycareservice.utils.HelperClass;
import com.app.citycareservice.utils.Params;
import com.google.android.material.button.MaterialButton;
import com.google.android.material.textfield.TextInputLayout;

import java.util.Objects;

public class EditProfileActivity extends AppCompatActivity implements Params {

    private static final String TAG = "EditProfileActivity";

    private Activity activity = EditProfileActivity.this;

    private ImageView back_iv;
    private TextInputLayout name_tif;
    private TextInputLayout email_tif;
    private TextInputLayout phone_tif;

    private MaterialButton update_btn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_edit_profile);

        back_iv = findViewById(R.id.back_iv);
        name_tif = findViewById(R.id.name_tif);
        email_tif = findViewById(R.id.email_tif);
        phone_tif = findViewById(R.id.phone_tif);
        update_btn = findViewById(R.id.update_btn);

        back_iv.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onBackPressed();
            }
        });

        update_btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                validateForm();
            }
        });

    }

    private void validateForm() {

        name_tif.setErrorEnabled(false);
        email_tif.setErrorEnabled(false);
        phone_tif.setErrorEnabled(false);

        String nameRegex = "^[A-Za-z]+$";

        String name = Objects.requireNonNull(name_tif.getEditText()).getText().toString().trim();
        String firstName = name;
        String lastName = "";
        String email = Objects.requireNonNull(email_tif.getEditText()).getText().toString().trim();
        String phone = Objects.requireNonNull(phone_tif.getEditText()).getText().toString().trim();

        if (TextUtils.isEmpty(name)) {
            name_tif.setError("Name Cannot be Empty");
            name_tif.setFocusable(true);
            return;
        }
        if (!name.matches(nameRegex)) {
            name_tif.setError("Invalid Name");
            name_tif.setFocusable(true);
            return;
        }

        if (TextUtils.isEmpty(email)) {
            email_tif.setError("Email Cannot be Empty");
            email_tif.setFocusable(true);
            return;
        }

        if (!Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
            email_tif.setError("Invalid Email Id");
            email_tif.setFocusable(true);
            return;
        }

        if (TextUtils.isEmpty(phone)) {
            phone_tif.setError("Phone Cannot be Empty");
            return;
        }
        if (Patterns.PHONE.matcher(phone).matches()) {
            phone_tif.setError("Invalid Phone NUmber");
            return;
        }

        Bundle bundle = new Bundle();
        bundle.putString(BUNDLE_KEY_USER_FIRST_NAME, firstName);
        bundle.putString(BUNDLE_KEY_USER_EMAIL, email);

//        if (isProfileImgSelected)
//            bundle.putString(BUNDLE_KEY_PROFILE_PIC_URI, profile_pic_uri.toString());
        if (!TextUtils.isEmpty(lastName))
            bundle.putString(BUNDLE_KEY_USER_LAST_NAME, lastName);

        startService(new Intent(activity, UpdateProfileService.class).putExtra(BUNDLE_NAME_PROFILE_DATA, bundle));
        HelperClass.showToast(activity, "Profile Update Successfully");
        finish();
    }
}

File: app/src/main/java/com/app/citycareservice/ui/activities/authentication/EditProfileActivity.kt
package com.app.citycareservice.ui.activities.authentication

import androidx.appcompat.app.AppCompatActivity
import android.app.Activity
import android.os.Bundle
import android.text.TextUtils
import android.content.Intent
import android.util.Patterns
import com.app.citycareservice.databinding.ActivityEditProfileBinding
import com.app.citycareservice.service.UpdateProfileService
import com.app.citycareservice.utils.HelperClass
import com.app.citycareservice.utils.Params
import com.app.citycareservice.utils.Params.BUNDLE_KEY_USER_EMAIL
import com.app.citycareservice.utils.Params.BUNDLE_KEY_USER_FULL_NAME
import com.app.citycareservice.utils.SharedPrefHandler

class EditProfileActivity : AppCompatActivity(), Params {
    private val activity: Activity = this@EditProfileActivity
    private lateinit var binding: ActivityEditProfileBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityEditProfileBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setData()

        binding.backIv.setOnClickListener { onBackPressed() }
        binding.updateBtn.setOnClickListener { validateForm() }
    }

    private fun setData() {
        val prefHandler = SharedPrefHandler(activity)
        with(binding) {
            nameTef.setText(prefHandler.getString(Params.SP_KEY_USER_NAME))
            emailTef.setText(prefHandler.getString(Params.SP_KEY_USER_EMAIL))
            phoneTef.setText(prefHandler.getString(Params.SP_KEY_USER_PHONE))
        }
    }

    private fun validateForm() {

        with(binding) {
            nameTif.isErrorEnabled = false
            emailTif.isErrorEnabled = false
            phoneTif.isErrorEnabled = false
            val fullName =
                nameTef.text.toString().trim()
            val email =
                emailTef.text.toString().trim()

            val bundle = Bundle()

            if (!TextUtils.isEmpty(fullName))
                bundle.putString(BUNDLE_KEY_USER_FULL_NAME, fullName)

            if (!TextUtils.isEmpty(email)) {
                if (!Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
                    emailTif.error = "Invalid Email Id"
                    emailTif.isFocusable = true
                } else
                    bundle.putString(BUNDLE_KEY_USER_EMAIL, email)
            }

            if (bundle.containsKey(BUNDLE_KEY_USER_FULL_NAME)
                || bundle.containsKey(BUNDLE_KEY_USER_EMAIL)
            ) {



                startService(
                    Intent(
                        activity, UpdateProfileService::class.java
                    ).putExtra(Params.BUNDLE_NAME_PROFILE_DATA, bundle)
                )
//                HelperClass.showToast(activity, "Profile Update Successfully")
                finish()
            } else {
                HelperClass.showToast(activity, "Please enter name or email to be updated")
            }
        }
    }

    companion object {
        private const val TAG = "EditProfileActivity"
    }
}